cf-agent/acl_posix.c

```diff
@@ -339,15 +341,33 @@ static int CheckPosixLinuxACEs(EvalContext *ctx, Rlist *aces, AclMethod method,
 
             if (!DONTDO)
             {
+                int last = -1;
+                acl_text_str = acl_to_any_text(acl_new, NULL, ',', 0);
+                Log(LOG_LEVEL_DEBUG, "ACL: new acl is `%s'", acl_text_str);
+
+                if ((retv = acl_check(acl_new, &last)) != 0)
+                {
+                    Log(LOG_LEVEL_ERR, "ACL: acl_check(%s): %s at entry %d. Refer to acl_check man page for validity rules.",
+			            acl_text_str,
+        			    acl_error(retv),
+        			    last);
+                    acl_free((void *) acl_existing);
```

Comment:
No casting for `void *` parameters, in C `void *` casting happens implicitly. It's a C++ thing that is not needed in C, and even considered harmful by some. I know the old code does it, but if possible change it.

Reply:
Do you have a reference for reading so I can get some background? Something like this? https://gustedt.wordpress.com/2014/04/02/dont-use-casts-i/

Reply:
First let me say that there are valid uses of casts. But when you use a cast you are basically telling the compiler "don't warn me of anything, I know what I'm doing". So in general it's good practice to keep the casts at a minimum and fix compiler warnings at their root cause, if possible (again, sometimes casts are needed).

The link you posted is quite good. There were also some more real problems with legacy C compilers: if you casted the return of a malloc() for example and had forgotten to include the proper header file, you might get a runtime error. Nice posts [here](https://stackoverflow.com/questions/1565496/specifically-whats-dangerous-about-casting-the-result-of-malloc).